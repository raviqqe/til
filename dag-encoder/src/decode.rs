mod decoder;

use crate::{decode::decoder::Decoder, Error, Program};

/// Decodes a program.
pub fn decode(codes: &[u8]) -> Result<Program, Error> {
    Decoder::new(codes).decode()
}
