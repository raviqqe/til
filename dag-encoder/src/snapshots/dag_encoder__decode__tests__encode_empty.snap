---
source: dag-encoder/src/decode.rs
expression: "decode([].as_slice())"
---
Err(
    MissingNode,
)
